### DISCUSSION TITLE
Please allow pinning my non-GitHub repositories to dev.to profile

### DISCUSSION THREAD

**Discussion by ell1e** [forem/forem#20209] [Feature Requests]
**Summed up: please allow pinning any public git/web-browsable repository URL on gitlab.com, SourceHut, and all other places and not just GitHub, to be pinned to a dev.to profile.**

**Original question:** How do I add repositories I work on to my dev.to profile, and where would i ask such a question? I couldn't find a help section on the page, and I saw profiles of other users that listed repositories under a dedicated subsection but I couldn't figure out how to do that myself. For what it's worth, I'm using e-mail-based login on dev.to, and my repositories are publicly hosted but on a GitHub.com/GitLab.com alternative so they wouldn't show up through these common OAuth2 integrations. They still definitely exist for real though, and I work on them, so listing them would be nice.

(Side note: neither the "Bio" nor the "Currently hacking on" profile sections seem to support links, so those fields don't seem to allow a good workaround either.)

**Comment by maestromac** [comment]
Navigate to the GitHub section on  https://dev.to/settings/extensions and you should the options to pin your public repos.

**Reply by ell1e** [reply]
I'm just seeing these sections: "Comment templates," "Publishing to DEV Community from RSS," "Web monetization," and "DEV Community API Keys."

**Reply by maestromac** [reply]
You'll need to connect to GitHub on https://dev.to/settings/account and then you should be able to pin repos.

**Reply by ell1e** [reply]
My repositories aren't on Github though!

**Reply by maestromac** [reply]
Ah forgive me, I missed that part. In that case, I'm sorry, GitHub is the only code repository we support for repo pinning.

**Reply by ell1e** [reply]
Couldn't you just allow specifying any repository URL? You could even add some manual delayed check through a moderator or something that probes to check whether that URL can be cloned, if you're so worried about abuse. (As for specifying someone else's repositories "maliciously" where I don't even contribute, I don't really see the problem, I could also link `apple.com` as my website or claim I'm good at some skill I know I'm not. Lying can't really be prevented and if anything, it gives the repository target free advertising anyway.)

**Reply by maestromac** [reply]
I understand that, but this is a rather old feature we have, and we don't have any plan to expand on it right now.

You are welcome to update this discussion to be a feature request or create a new one.

**Reply by ell1e** [reply]
It's a nice feature so I sincerely hope you consider updating it to allow entering any repository hosted in a common web-browsable and git clonable way on any of the public-facing platforms for this, where many more than GitHub exist by now.

To mitigate the impact, allowing URLs in "Currently hacking on" to be actually clickable would be a start.

**Comment by ell1e** [comment]
I'm not familiar with discussions, why does this one show up as "closed"?

**Reply by maestromac** [reply]
I've prematurely closed it. I've re-opened it now.
